src/Bicep.Core.IntegrationTests/DynamicAzTypesTests.cs

```diff
@@ -67,5 +72,16 @@ public async Task Az_namespace_errors_with_configuration()
                 ("BCP205", DiagnosticLevel.Error, "Provider namespace \"az\" does not support configuration."),
             });
         }
+
+        [TestMethod]
+        public async Task Az_namespace_can_be_used_with_alias()
+        {
+            var services = await GetServices();
+            var result = await CompilationHelper.RestoreAndCompile(services, @$"
+            provider 'br/public:az@{BicepTestConstants.BuiltinAzProviderVersion}' as testAlias
+            ");
+
+            result.Should().GenerateATemplate();
```

Comment:
Could be worth checking here that the symbol was bound as expected. I think that would look something like `result.Compilation.GetEntrypointSemanticModel().Root.ProviderDeclarations.Should().Contain(x => x.Name.Equals("testAlias"))`

Reply:
yeah, that is a good catch! Adding this assertion as well